src/linker/Linker/XmlCustomAttributeSource.cs

```diff
@@ -362,66 +391,54 @@ void ProcessMethodSignature (TypeDefinition type, string signature, XPathNodeIte
-					if (returnparamAnnotation != 0)
-						returnAnnotation = returnparamAnnotation;
-				}
-			} else {
-				_context.LogMessage (MessageContainer.CreateWarningMessage (_context,
-					$"There is more than one return parameter specified for '{method.Name}' in '{_xmlDocumentLocation}'", 2023, _xmlDocumentLocation));
-			}
-			if (returnAnnotation != 0 || parameterAnnotation.Count > 0)
-				_methods[method] = new AnnotatedMethod (returnAnnotation, parameterAnnotation.ToArray ());
-
+			IEnumerable<CustomAttribute> attributes = ProcessAttributes (iterator.Current);
+			if (attributes.Count () > 0)
+				_attributes[method] = attributes;
+			ProcessReturnParameters (method, iterator);
+			ProcessParameters (method, iterator);
 		}
 
-		ArrayBuilder<(string, ArrayBuilder<Attribute>)> ProcessParameters (TypeDefinition type,
-			MethodDefinition method, XPathNodeIterator iterator)
+		void ProcessParameters (MethodDefinition method, XPathNodeIterator iterator)
 		{
-			var methodParameters = new ArrayBuilder<(string, ArrayBuilder<Attribute>)> ();
+			iterator = iterator.Current.SelectChildren ("parameter", string.Empty);
 			while (iterator.MoveNext ()) {
-				methodParameters.Add ((GetAttribute (iterator.Current, "name"), ProcessAttributes (iterator)));
+				IEnumerable<CustomAttribute> attributes = ProcessAttributes (iterator.Current);
+				if (attributes.Count () > 0) {
+					string paramName = GetAttribute (iterator.Current, "name");
+					foreach (ParameterDefinition parameter in method.Parameters) {
+						if (paramName == parameter.Name) {
+							if (_attributes.ContainsKey (parameter))
+								_context.LogMessage (MessageContainer.CreateWarningMessage (_context, $"There are duplicate parameter names for '{paramName}' inside '{method.Name}' in '{_xmlDocumentLocation}'", 2024, _xmlDocumentLocation));
+							_attributes[parameter] = attributes;
+							break;
+						}
+					}
+				}
 			}
-			return methodParameters;
 		}
 
-		ArrayBuilder<ArrayBuilder<Attribute>> ProcessReturnParameters (TypeDefinition type,
-			MethodDefinition method, XPathNodeIterator iterator)
+		void ProcessReturnParameters (MethodDefinition method, XPathNodeIterator iterator)
 		{
-			var methodParameters = new ArrayBuilder<ArrayBuilder<Attribute>> ();
-
+			iterator = iterator.Current.SelectChildren ("return", string.Empty);
+			bool firstAppearance = true;
 			while (iterator.MoveNext ()) {
-				methodParameters.Add (ProcessAttributes (iterator));
+				if (firstAppearance) {
+					firstAppearance = false;
+					IEnumerable<CustomAttribute> attributes = ProcessAttributes (iterator.Current);
+					if (attributes.Count () > 0)
+						_attributes[method.MethodReturnType] = attributes;
+				} else {
+					_context.LogMessage (MessageContainer.CreateWarningMessage (_context, $"There is more than one return parameter specified for '{method.Name}' in '{_xmlDocumentLocation}'", 2023, _xmlDocumentLocation));
```

Comment:
Why is it warning?

Reply:
Same as above - two values for the same thing - we will intentionally drop data in this case so we should warn about it.